jparest/src/main/java/uk/gov/homeoffice/digital/sas/jparest/repository/JpaRestRepositoryImpl.java

```diff
@@ -0,0 +1,22 @@
+package uk.gov.homeoffice.digital.sas.jparest.repository;
+
+import java.io.Serializable;
+import javax.persistence.EntityManager;
+import org.springframework.data.jpa.repository.support.SimpleJpaRepository;
+
+
+/**
+ * Repository that queries entities and has common functionality for paging, sorting
+ * and filtering resources.
+ * <p>
+ * Resources for ManyToMany relationships can also be queried.
+ */
+public class JpaRestRepositoryImpl<T, Y extends Serializable>
```

Comment:
A couple of questions/considerations re: design of the repositories:
- Would the `JpaRestRepository` interface, added above (with `@NoRepositoryBean` annotation), have been necessary, have the implementation class (`JpaRestRepositoryImpl`) been instead an interface, extending Spring's `JpaRestRepository<T extends BaseEntity, Y extends Serializable>` ?
- How would this repository class/interface co-exist with the client code repository interfaces (e.g. a TimeEntryRepository has been added in timecard-restapi) ?

I appreciate there is a separate sub-task to refactor repositories (https://collaboration.homeoffice.gov.uk/jira/browse/EAHW-2543). So the questions above can be addressed part of that sub-task

Reply:
- So the reason I had to use a class is because there will be actual implementation of methods for the queries. E.g. all the criteria builder queries currently in the service. And they're exposed by our interface which then extends the built in JpaRepository so that we also get the built in stuff to use. I appreciate that intention isn't clear in this PR as currently the repo is a placeholder.
- In theory it should work seamlessly as technically from the client's perspective nothing has really changed. Only the inner implementation which is instead of building & executing queries in the service it now happens in a separate layer. 
  - But yes this will me more of a question and definitely something to test in the next subtask.